contractcourt/channel_arbitrator.go

```diff
@@ -959,6 +959,30 @@ func (c *ChannelArbitrator) stateStep(
 			return StateDefault, closeTx, nil
 		}
 
+		// Cancel all the outgoing dust htlcs available either on the
```

Comment:
I think instead of using `c.checkLocalChainActions`, we should instead use `constructChainActions` here, then call `resolveHTLCsImmediately`, maybe we can break `resolveHTLCsImmediately` into two methods, one for cancel the breach case, one for the rest - need to think twice hmm

Reply:
I am not sure why you think we can call `resolveHTLCsImmediately ` because the commitment is still not confirmed here ?

Reply:
because we don't do non-dust canceling here. With the current design, we are almost sure we will cancel the dust twice - first time here, second time when the FC tx is confirmed, which I think is unnecessary.

Reply:
I actually think it's better to sperate the canceling of the non-dust vs dust, it's more clear and avoid duplication. For dust HTLCs, which is what `resolveHTLCsImmediately` does, we can just do it here, with renaming it to `resolveDustHTLCs`. Then in a future PR where we decide to cancel non-dust HTLCs, we can do it under `StateContractClosed`.

Reply:
ok changed, we are now canceling dust only once.